src/reducers/ui.js

```diff
@@ -85,6 +97,30 @@ export default (state = EMPTY_UI, action) => {
                 input: action.value,
             }
         }
+        case SET_UI_PROGRAM: {
+            return {
+                ...state,
+                program: action.value,
+            }
+        }
+        case SET_UI_PROGRAM_ID: {
+            return {
+                ...state,
+                program: {
+                    ...state.program,
+                    id: action.value,
+                },
+            }
+        }
+        case SET_UI_PROGRAM_STAGE: {
+            return {
+                ...state,
+                program: {
+                    ...state.program,
+                    stage: action.value,
+                },
+            }
+        }
```

Comment:
In the issue I am currently working on ([TECH-876](https://jira.dhis2.org/browse/TECH-876)), I need to  show a select from which a user can select a program stage (for tracker programs). To populate this select with options, I was thinking of including the `programStages` field when I GET the `programs`. If I go ahead with this we will end up with a `program.stage` and a `program.programStages` property. I would find this a little confusing. Instead we could perhaps only persist the ID and make the name very explicit, i.e. change `stage` (object) to `selectedStageId` (string).